packages/framework/src/Form/Transformers/CopyTotalPricesOfOrderItemTransformer.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Form\Transformers;
+
+use Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+final class CopyTotalPricesOfOrderItemTransformer implements DataTransformerInterface
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData|null $value
+     * @return \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData|null
+     */
+    public function transform($value): ?OrderItemData
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if ($value instanceof OrderItemData) {
```

Comment:
This block could be simplified to more readable:

```php
if (!$value instanceof OrderItemData) {
    throw new TransformationFailedException(...);
}

if ($value->quantity !== 1) {
    throw new TransformationFailedException(sprintf('A single item must provided for total prices to be copiable, quantity of %d provided.', $value->quantity));
}

$value->totalPriceWithVat = $value->priceWithVat;
$value->totalPriceWithoutVat = $value->priceWithoutVat;

return $value;
```
the advantage is less nested IFs

Reply:
You're right about that, fixed in e3e298f20.

Also, I just found out that `instanceof` has a [higher operator precendence](https://www.php.net/manual/en/language.operators.precedence.php) than negation... I always wrote `if (!($object instanceof ClassName)) {` and I was annoyed by how ugly it looks. Learning something every day :smile: Thanks!

Reply:
the has higher operator precedence.... it's like that since... ever (2004 I guess)